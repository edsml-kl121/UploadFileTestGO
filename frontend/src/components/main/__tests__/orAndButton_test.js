import {render, screen, cleanup} from "@testing-library/react"
import OrButton from "../helpercomponent/orAndButton.component.js";
import renderer from 'react-test-renderer'

afterEach(() => {
  cleanup()
})


test('matches snapshot', () => {
  const tree = renderer.create(<OrButton/>)
  expect(tree).toMatchSnapshot();
})

test('should render or component', () => {
  render(<OrButton/>)
  const buttonElement = screen.getByTestId('or-1')
  expect(buttonElement).toBeInTheDocument();
  expect(buttonElement).toHaveTextContent('OR');
})


test('should render have a button enabled', () => {
  render(<OrButton/>)
  const button = screen.getByTestId('button-1')
  expect(button).not.toBeDisabled();
})