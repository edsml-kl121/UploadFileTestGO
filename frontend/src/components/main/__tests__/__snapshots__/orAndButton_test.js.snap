// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-lg-3 col"
    />
    <div
      className="col-lg-6 col"
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "#96A1BB",
              "flex": 1,
              "height": "2px",
            }
          }
        />
        <div>
          <p
            data-testid="or-1"
            style={
              Object {
                "color": "#96A1BB",
                "marginTop": "12px",
                "textAlign": "center",
                "width": "70px",
              }
            }
          >
            OR
          </p>
        </div>
        <div
          style={
            Object {
              "backgroundColor": "#96A1BB",
              "flex": 1,
              "height": "2px",
            }
          }
        />
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained px-5 py-3 MuiButton-contained px-5 py-3Primary MuiButton-sizeMedium MuiButton-contained px-5 py-3SizeMedium MuiButton-root MuiButton-contained px-5 py-3 MuiButton-contained px-5 py-3Primary MuiButton-sizeMedium MuiButton-contained px-5 py-3SizeMedium css-1w1rijm-MuiButtonBase-root-MuiButton-root"
        data-testid="button-1"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "#6749F5",
            "boxShadow": "none",
            "color": "white",
            "fontSize": "32px",
            "textTransform": "unset",
          }
        }
        tabIndex={0}
        type="button"
      >
        Browse File
      </button>
    </div>
    <div
      className="col-lg-3 col"
    />
  </div>
</div>
`;
